### DISCUSSION TITLE
General availability for problem solution

### DISCUSSION THREAD

**Discussion by agoloborodko** [Codium-ai/AlphaCodium#30] [Q&A] [answered]
Hi Team!

Your solution is impressive, yet I wasn't able to find any developer-fiendly guide for running AlphaCodium to solve generic problems, which are not included to your own datasets.

Do you have any plans to add support for prompting arbitrary code generating problems in the future?

Thanks!

**Comment by mrT23** [comment, marked as answer]
There is a lengthy section in the readme that talks about "Broader Applicability":
https://github.com/Codium-ai/AlphaCodium?tab=readme-ov-file#broader-applicability

I think that when coupled with the actual code implementation in this repo, these tips do provide a viable way to take _parts_ of AlphaCodium solution and apply them to more general problems (bullet points analysis, semantic reasoning, generating modular code and so on...)
